Feat pattern validator

takrishna opened the pull request:
1. Added IsPattern validator for strings validators
2. Modified - Default Translation files for the error message
3. Added tests 

#My first attempt to contribute - #caution

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ing-bank/lion?pullRequest=719) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ing-bank/lion?pullRequest=719) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ing-bank/lion?pullRequest=719) it.</sub>

tlouisse commented:
Looks good! 
Nitpicking: I think the name `isPattern` is not completely correct, because you test the modelValue _against_ a pattern instead of testing whether the modelValue _is_ a pattern.

I think just `Pattern` would be a good name for this validator.

takrishna commented:
Agreed, "Pattern" is simple and easy - I was originally considering "IsAPatternOf" to be verbose .. phew.. !

tlouisse commented on packages/validate/src/validators/StringValidators.js:
```diff
@@ -89,3 +89,19 @@ export class IsEmail extends Validator {
   }
 }
+
+const isPattern = (value, pattern) => pattern.test(value);
```
And one more nitpick: maybe call this `matchesPattern` or `hasPattern` then?

tlouisse commented on packages/validate/test/StringValidators.test.js:
```diff
@@ -111,4 +112,16 @@ describe('String Validation', () => {
+  it('provides new Pattern() to allow only valid patterns', () => {
+    let isEnabled;
+    const validator = new Pattern(/#LionRocks/);
+    expect(validator.constructor.name).to.equal('Pattern');
```
We should test for `validator.constructor.validatorName`.
(We probably do this wrong in the other tests as well (we couldn't use `constructor.name` because it gets lost in code minification/transpilation) )

gerjanvangeest commented on packages/validate/translations/it.js:
```diff
@@ -6,6 +6,7 @@ export default {
     MaxLength: 'Inserire un(a) {fieldName} corretto(a) (fino a {params} caratteri).',
     MinMaxLength:
       'Inserire un(a) {fieldName} corretto(a) (tra {params.min} e {params.max} caratteri).',
+    Pattern: 'Kérjük, adjon meg érvényes {fieldName} értéket.',
```
I believe you copied here the Hungarian translation, instead of the Italian `IsNumber`

takrishna replied:
trust me I am better than this :D - fixed

gerjanvangeest commented on packages/validate/translations/sk.js:
```diff
@@ -5,6 +5,7 @@ export default {
     MinLength: 'Uveďte správne {fieldName} (najmenej {params}).',
     MaxLength: 'Uveďte správne {fieldName} (maximálne {params} znakov).',
     MinMaxLength: 'Uveďte správne {fieldName} ({params.min} až {params.max} znakov).',
+    Pattern: 'Введите действительное значение поля {fieldName}.',
```
I believe this is Russian

tlouisse commented:
I think if you fix the suggestions from @gerjanvangeest and put everything in one commit, it will be good to merge 💪

takrishna commented:
> I think if you fix the suggestions from @gerjanvangeest and put everything in one commit, it will be good to merge 💪

Fixed and Squashed into one commit

daKmoR commented:
mind you adding a throw later is a breaking change 😅

while removing a throw is find... so better add now then later... 🤔

jorenbroekema commented:
Thanks for contributing :)!
